server/modules/record/manager/recordManager.js

```diff
@@ -28,21 +27,21 @@ export const insertRecord = async (user, surveyId, record, system = false, clien
     return recordDb
   })
 
-export const insertNodesFromValues = async (user, surveyId, nodeValues, client = db) => {
-  const activities = nodeValues.map((nodeValuesRow) => {
-    const node = NodeRepository.tableColumns.reduce(
-      (accContent, key, index) => Object.assign(accContent, { [camelize(key)]: nodeValuesRow[index] }),
-      {}
-    )
-    return ActivityLog.newActivity(ActivityLog.type.nodeCreate, node, true)
-  })
-
-  await client.tx(async (t) =>
-    Promise.all([
-      NodeRepository.insertNodesFromValues(surveyId, nodeValues, t),
-      ActivityLogRepository.insertMany(user, surveyId, activities, t),
-    ])
-  )
+export const insertNodesInBulk = async ({ user, surveyId, nodes }, tx) => {
+  const nodeValues = nodes.map((node) => [
+    Node.getUuid(node),
+    Node.getDateCreated(node),
+    Node.getDateCreated(node),
+    Node.getRecordUuid(node),
+    Node.getParentUuid(node),
+    Node.getNodeDefUuid(node),
+    JSON.stringify(Node.getValue(node, null)),
+    Node.getMeta(node),
+  ])
+  const activities = nodes.map((node) => ActivityLog.newActivity(ActivityLog.type.nodeCreate, node, true))
+
+  await NodeRepository.insertNodesFromValues(surveyId, nodeValues, tx)
+  await ActivityLogRepository.insertMany(user, surveyId, activities, tx)
```

Comment:
is responsibility of this manager?

Reply:
uhm, usually we use the repositories and the activity log repository in the manager modules...